### PR TITLE
Dynamically Created Accordions Not Opening

### LINKED ISSUE #669: Dynamically added panels in go-accordion are not opening.
**Describe the bug**
In go-accordion the panels added dynamically are not opening. 

**To Reproduce**
https://stackblitz.com/edit/goaccordion

**Expected behavior**
All panels visible in the accordion should open.

### PR DESCRIPTION
**Pull request by StevenUlmer**
## PR Checklist
Please check if your PR fulfills the following requirements:
- [x] The commit message follows our [guidelines](https://github.com/mobi/goponents/blob/master/CONTRIBUTING.md)
- [x] Tests for the changes have been added
- [x] Docs have been added or updated

## PR Type
What kind of change does this PR introduce?
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Style Update (CSS)
- [ ] Other... Please describe:

## What is the current behavior?
Resolves #669 

## What is the new behavior?
Dynamically created accordions function the same as html driven accordions

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/go-lib/src/lib/components/go-accordion/go-accordion.component.ts [hunk 1]
```diff
@@ -41,24 +42,27 @@ export class GoAccordionComponent implements OnInit, AfterContentInit, OnDestroy
   }
 
   ngAfterContentInit(): void {
-    this.panels.toArray().forEach((panel: GoAccordionPanelComponent, index: number) => {
-      this.updatePanelState(panel, index);
-      this.subscribePanel(panel);
-      panel.detectChanges();
+    this.updateChildPanels();
+
+    this.panels.changes.pipe(takeUntil(this.destroy$)).subscribe(() => {
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-accordion/go-accordion.component.ts:47]: Should we look at adding these to a local `subscriptions` variable and unsubscribe from them on destroy?

File: projects/go-lib/src/lib/components/go-accordion/go-accordion.component.ts [hunk 2]
```diff
@@ -41,24 +42,27 @@ export class GoAccordionComponent implements OnInit, AfterContentInit, OnDestroy
...
+    this.panelsDestroy$.complete();
   }
   //#endregion
 
   //#region Private Methods
   /////////////////////////
 
   private subscribePanel(panel: GoAccordionPanelComponent): void {
-    panel.toggle.pipe(takeUntil(this.destroy$)).subscribe(() => {
+    panel.toggle.pipe(takeUntil(this.panelsDestroy$)).subscribe(() => {
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-accordion/go-accordion.component.ts:65]: Same question here, should we add this to a description?

**Review by grahamhency — COMMENTED** [review]
Just on question on subscriptions

File: projects/go-lib/src/lib/components/go-accordion/go-accordion.component.ts (hunk 1 above)
Comment (StevenUlmer) [reply projects/go-lib/src/lib/components/go-accordion/go-accordion.component.ts:47]: The `takeUntil(this.destroy$)` completes the subscription when the component is destroyed thus cleaning up any outstanding memory leaks.

**Review by grahamhency — APPROVED** [review]

**Pull request merged**